src/main.c

```diff
@@ -301,34 +301,46 @@ static int get_character_at_grid(char* board, int x, int y)
 	return board[posy*9+posx];
 }
 
-static void fill_grid(char *board, int x_cursor, int y_cursor)
+static void fill_grid(char *user_board, char *plain_board, int x_cursor, int y_cursor)
 {
 	int row, col, x, y;
 	int n;
 	int c;
 	int selected;
+	int isUserInput;
+	int m;
 
-	selected = get_character_at_grid(board, x_cursor, y_cursor);
+	selected = get_character_at_grid(user_board, x_cursor, y_cursor);
 	wstandend(grid);
 	y = GRID_NUMBER_START_Y;
 	for(row=0; row < 9; row++)
 	{
 		x = GRID_NUMBER_START_X;
 		for(col=0; col < 9; col++)
 		{
-			n = board[row*9+col];
+		        isUserInput = 1; // True
+			n = user_board[row*9+col];
 			if(n == '.')
 				c = ' ';
 			else
+			{
 				c = n;
+				m = plain_board[row*9+col];
+				if(n == m)
+				        isUserInput = 0; // False
+			}
 			if (g_useColor && g_useHighlights && selected == c)
 			{
 				if (x == x_cursor && y == y_cursor)
 					wattron(grid, COLOR_PAIR(COLOR_HIGHLIGHT_CURSOR));
 				else
 					wattron(grid, COLOR_PAIR(COLOR_HIGHLIGHT));
 			}
+			if(isUserInput)
+			        wattron(grid, COLOR_PAIR(3));
```

Comment:
wrong indentation.

Reply:
Thanks for adapting the PR.
In case you use vim it should automatically take the right settings because of the vim header at the top of the file :)
All the source code uses tabs instead of whitespaces for indentation.
If you compare this `if(isUserInput)` block with the next one you see that indentation is different there. It should be the same.
If you use vim `!set list` will make the tabs/whitespaces visible.

Reply:
Ah! You use vim with tabs and I use emacs with spaces (I guess we can't be friends). I've done my best to ensure everything is fixed up now in 87f981c259c16a34d886f14c101a02e1242d5b9a